File: app/pages/repos/_repo_id/models/_model_id/production_predictions/predictions/_identifier/props.ts
import { InputTable, Prediction } from "common/prediction_result"
import { ModelLayoutInfo } from "layouts/model_layout"

export type Props = {
	identifier: String
	inner: Inner
	modelLayoutInfo: ModelLayoutInfo
}

export enum InnerType {
	Found = "found",
	NotFound = "not_found",
}

export type Inner =
	| {
			type: InnerType.Found
			value: Found
	  }
	| {
			type: InnerType.NotFound
			value: null
	  }

export type Found = {
	date: String
	inputTable: InputTable
	prediction: Prediction
}

File: app/pages/repos/_repo_id/models/_model_id/production_predictions/predictions/_identifier/props.rs
use tangram_app_common::predict::{InputTable, Prediction};
use tangram_app_layouts::model_layout::ModelLayoutInfo;

#[derive(serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Props {
	pub model_layout_info: ModelLayoutInfo,
	pub identifier: String,
	pub inner: Inner,
}

#[derive(serde::Serialize)]
#[serde(rename_all = "snake_case")]
#[serde(tag = "type", content = "value")]
pub enum Inner {
	NotFound,
	Found(Found),
}

#[derive(serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Found {
	pub date: String,
	pub input_table: InputTable,
	pub prediction: Prediction,
}
